rosetta-lang/src/main/java/com/regnosys/rosetta/types/TypeSystem.java

```diff
@@ -95,6 +96,28 @@ private RType getRulesInputType(RDataType data, Optional<RosettaExternalRuleSour
     private RType getRulesInputTypeFromCache(RDataType data, Optional<RosettaExternalRuleSource> source, Provider<RType> typeProvider) {
     	return cache.get(new Pair<>(RULE_INPUT_TYPE_CACHE_KEY, new Pair<>(data, source)), typeProvider);
     }
+    
+    public RMetaAnnotatedType join(RMetaAnnotatedType t1, RMetaAnnotatedType t2) {
```

Comment:
For consistency, I would call this one `joinMetaAnnotatedTypes` as well.

Reply:
Done